src/Bearded.TD/Game/Simulation/Elements/ElementalPhenomenonScopeBase.cs

```diff
@@ -92,36 +87,44 @@ private void startEffect(TEffect effect, out ElementalStatus? newStatus)
 
     private void endEffectIfPreviouslyActive()
     {
-        if (activeEffect is null) return;
+        if (activeEffect is not { } current) return;
 
-        EndScope(target);
-        activeEffect.StatusIcon?.DeleteImmediately();
+        EndScope(current.Effect);
+        current.StatusIcon?.DeleteImmediately();
         activeEffect = null;
     }
 
     private void updateStatusIconExpiry(IStatusReceipt statusIcon)
     {
-        var latestExpiry = activeEffects.Select(e => e.Expiry).Max();
+        var latestExpiry = activeEffects.Max(static e => e.Expiry);
         statusIcon.SetExpiryTime(latestExpiry);
     }
 
-    protected abstract bool TryChooseEffect(out TEffect effect);
-    protected abstract void StartScope(GameObject target, out ElementalStatus? status);
-    protected abstract void StartEffect(GameObject target, TEffect effect, EffectStartContext context);
-    protected abstract void ApplyEffectTick(GameObject target, TEffect effect);
-    protected abstract void EndEffect(GameObject target, TEffect effect);
-    protected abstract void EndScope(GameObject target);
+    protected abstract TEffect? TryChooseEffect(ReadOnlySpan<EffectWithExpiry> activeEffects);
+    protected abstract void StartScope(TEffect effect, ref EffectChangeResult? statusChange);
+    protected abstract void ChangeActiveEffect(TEffect previousEffect, TEffect newEffect, ref EffectChangeResult? statusChange);
+    protected abstract void ApplyEffectTick(TEffect effect);
+    protected abstract void EndScope(TEffect effect);
+
+    protected readonly record struct EffectWithExpiry(TEffect Effect, Instant Expiry);
+    private readonly record struct ActiveEffect(TEffect Effect, IStatusReceipt? StatusIcon);
 
-    private readonly record struct EffectWithExpiry(TEffect Effect, Instant Expiry);
-    private sealed record ActiveEffect(TEffect Effect, IStatusReceipt? StatusIcon);
+}
 
-    protected class EffectStartContext
+// Base class for non-generic members
+abstract class ElementalPhenomenonScopeBase
+{
+    public readonly struct EffectChangeResult
     {
-        public ElementalStatus? NewStatus { get; private set; }
+        public ElementalStatus? NewStatus { get; private init; }
 
-        public void ChangeStatus(ElementalStatus status)
-        {
-            NewStatus = status;
-        }
+        public static EffectChangeResult HideStatus => default;
+
+        public static EffectChangeResult ShowStatus(ElementalStatus status) => new() { NewStatus = status };
+
+        public static implicit operator EffectChangeResult(ElementalStatus newStatus) => ShowStatus(newStatus);
     }
+
+    protected static EffectChangeResult HideStatus => EffectChangeResult.HideStatus;
```

Comment:
nit: I think I'd rather solve this with static imports. Somewhat subjective perhaps, but this feels like a weird thing to have as a property.